src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,347 @@
+                payment_type: paymentData.tokenizePayload.type,
+                nonce: paymentData.tokenizePayload.nonce,
+                provider: 'googlepay',
+                action: 'set_external_checkout',
+                card_information: this._getCardInformation(cardInformation),
+            }),
+        });
+    }
+
+    private _getCardInformation(cardInformation: { cardType: string, lastFour: string }) {
+        return {
+            type: cardInformation.cardType,
+            number: cardInformation.lastFour,
+        };
+    }
+
+    private _handleError(error: Error): never {
+        throw new StandardError(error.message);
+    }
+
+    private _getPayment() {
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(this._methodId))
+            .then(() => {
+                const state = this._store.getState();
+                const paymentMethod = state.paymentMethods.getPaymentMethod(this._methodId);
+
+                if (!paymentMethod) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                if (!paymentMethod.initializationData.nonce) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPayment);
+                }
+
+                const paymentData = {
+                    method: this._methodId,
+                    nonce: paymentMethod.initializationData.nonce,
+                    cardInformation: paymentMethod.initializationData.card_information,
+                };
+
+                this._paymentMethod = paymentMethod;
+
+                return {
+                    methodId: this._methodId,
+                    paymentData,
+                };
+            });
+    }
+
+    private _updateShippingAndBillingAddress(paymentSucessPayload: PaymentSuccessPayload): Promise<void> {
+        return Promise.all([
+                this._synchronizeShippingAddress(paymentSucessPayload.shippingAddress),
+                this._synchronizeBillingAddress(paymentSucessPayload.billingAddress),
+            ]).then(() => Promise.resolve());
+    }
+
+    private _synchronizeShippingAddress(shippingAddress: GooglePayAddress): Promise<InternalCheckoutSelectors | void> {
+
+        if (!this._methodId) {
+            throw new RemoteCheckoutSynchronizationError();
+        }
+
+        if (!shippingAddress) {
+            return Promise.resolve();
+        }
+
+        return this._store.dispatch(
+            this._consignmentActionCreator.updateAddress(mapGooglePayAddressToRequestAddress(shippingAddress))
+        ).then(() => this._store.getState());
+    }
+
+    private _synchronizeBillingAddress(billingAddress: GooglePayAddress): Promise<InternalCheckoutSelectors> {
+        if (!this._methodId) {
+            throw new RemoteCheckoutSynchronizationError();
+        }
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(this._methodId))
```

Comment:
Yes because we don't save in any local variable de actual billingAddress. and we need to get the `billingAddress.id` to update (if needed) the billing address.

Reply:
The payment method object shouldn't contain any information about the billing address? So why do you need to load it the payment method?

Reply:
Actually payment method object contains `state.billingAddress.getBillingAddress()` and I used to get the `remoteBillingAddress.id` to update the actual billingAddress